### PR TITLE
make it work

### PR DESCRIPTION
**Pull request by d-ryzhykau**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,62 +5,71 @@
...
-}
+const startConveyor = () => {
+    const conveyorState = Array(STAGES.length).fill(null);
+    const storageState = ITEMS.slice();
+    setInterval(() => {
+        console.log("conveyor state", conveyorState);
+        console.log("storage state", storageState);
+        // out
+        const output = conveyorState[conveyorState.length - 1];
+        if (output != null) {
```
Comment (olga-arkhipenko) [inline index.js:48]: oh please i'm begging u `!==`

File: index.js [hunk 2]
```diff
@@ -5,62 +5,71 @@
...
+          }
+        }
 
-const startConveyor = () => {
-    const items = document.querySelectorAll('.item');
-    items.forEach(item => manufacture(item));
-};
+        // in
+        const input = storageState.shift();
+        if (input != null) {
```
Comment (olga-arkhipenko) [inline index.js:66]: u r gansta, ye?

File: index.js [hunk 3]
```diff
@@ -5,62 +5,71 @@
...
+          STAGES[STAGES.length - 1].removeChild(output);
+          storageState.push(output);
+          STORAGE.appendChild(output);
+        }
 
+        // next stage
+        for (let i = STAGES.length - 1; i > 0; --i) {
+          const itemMoved = conveyorState[i - 1];
+          conveyorState[i] = itemMoved;
+          if (itemMoved != null) {
```
Comment (olga-arkhipenko) [inline index.js:58]: please no

File: index.js [hunk 4]
```diff
@@ -5,62 +5,71 @@
...
+        const input = storageState.shift();
+        if (input != null) {
+          STORAGE.removeChild(input);
 
+          conveyorState[0] = input;
+          STAGES[0].appendChild(input);
+        }
+    }, 3000);
+};
+document.querySelector('.start-button').onclick = startConveyor;
```
Comment (olga-arkhipenko) [inline index.js]: nice!
and we can make it even better using `addEventListener` instead

**Review by olga-arkhipenko — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (d-ryzhykau) [reply index.js:48]: `console.assert(null == undefined)`

File: index.js (hunk 2 above)
Comment (d-ryzhykau) [reply index.js:66]: image

**Pull request closed without merging**